src/dispatch/report/models.py

```diff
@@ -0,0 +1,63 @@
+from datetime import datetime
+
+from typing import List, Optional
+
+from sqlalchemy import Column, DateTime, Integer, String, ForeignKey, event
+from sqlalchemy.orm import relationship
+from sqlalchemy_utils import JSONType, TSVectorType
+
+
+from dispatch.database import Base
+from dispatch.models import DispatchBase
+
+from .enums import ReportTypes
+
+
+class Report(Base):
+    id = Column(Integer, primary_key=True)
+    created_at = Column(DateTime, default=datetime.utcnow)
+    details = Column(JSONType, nullable=True)
+    details_raw = Column(String, nullable=True)
+    type = Column(String, nullable=False, server_default=ReportTypes.tactical_report)
+
+    # relationships
+    incident_id = Column(Integer, ForeignKey("incident.id"))
+    participant_id = Column(Integer, ForeignKey("participant.id"))
+    document = relationship("Document", uselist=False, backref="report")
+
+    # full text search capabilities
+    search_vector = Column(TSVectorType("details_raw"))
+
+    @staticmethod
+    def _details_raw(mapper, connection, target):
+        target.details_raw = "; ".join(
+            [f"{k.capitalize()}: {v}" for k, v in target.details.items()]
```

Comment:
I'm not sure if it will affect searching, but I would rather we just concat all the raw strings together, i.e. if I search for `conditions` it shouldn't return every tactical report.

Reply:
`details_raw` will contain something like this, for example: `Conditions: These are the conditions; Actions: These are the actions; Needs: These are the needs`. Isn't that what we want?

Reply:
Yes, but if you include the word `condition` if I search for `condition` all status reports will be returned (since you've included that word) even though it never showed up in the original condition text. Additionally, that column will never be used for anything other than indexing so we don't really need to add that metadata.

Reply:
Ah, got it. I'll remove the key.